### PR TITLE
Register Lambda Context as singletons and MDC for every Handler

### PR DESCRIPTION
**Pull request by theaspect**
Fixes: https://github.com/micronaut-projects/micronaut-aws/issues/1332
Fixes: https://github.com/micronaut-projects/micronaut-aws/issues/1326

This is first attempt to generalize adding diagnostic information to lambda handler despite significant difference between `MicronautRequestHandler` and `MicronautLambdaHandler`. See also #1287 

I'll update documentation and add test when receive a green light for choosen approach.

@graemerocher and @sdelamo your comments are welcome especially regarding bean initialization and registration.

### CODE REVIEW

File: function-aws/src/main/java/io/micronaut/function/aws/DiagnosticInfoPopulator.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import io.micronaut.context.ApplicationContext;
+import io.micronaut.core.annotation.Introspected;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.util.StringUtils;
+import org.slf4j.MDC;
+
+import java.util.Optional;
+
+@Introspected
+public class DiagnosticInfoPopulator {
```
Comment (graemerocher) [inline function-aws/src/main/java/io/micronaut/function/aws/DiagnosticInfoPopulator.java:15]: not sure why this is `@Introspected`? It doesn't have any properties

**Comment by graemerocher** [comment]
General approach seems fine

**Comment by sdelamo** [comment]
@theaspect thanks for the contribution. I will review it during this week and gets this merged.

**Comment by sdelamo** [comment]
I have created two beans: 

`DiagnosticInfoPopulator`
`LambdaContextFactory` 

so that users can provide their own implementations if what we do is not what they are looking for. 

These PR tests that  Lambda `Context` beans are registered. 

It also tests how to inject the Lambda `Context`. I added a docs section.

**Review by graemerocher — APPROVED** [review]

**Pull request merged**